### ISSUE TITLE
[RAV-11] Add support for chat in Desk interface

### ISSUE THREAD

**Issue by ajiragroup** [The-Commit-Company/Raven#502]
You guys did a great job here with the app. but i feel it not quite practical. 

Reasons ? Only one.
1. You need to open a separate interface to use the app even if using along with erpnext.

Solution:
1. This should also function as a chat pop up like facebook messenger and also an option to use a full window app (as it is right now) does not hurt.

This is because, we do our important stuffs in erpnext - the main app. If we would get a notification about incoming message in the erpnext desk itself, it would be better. Just open up the popup, reply/read and continue with the app.

So, basically, I am requesting for a feature : create an option to use it as a chat app like frappe chat along with all your awesome features.

<sub>[RAV-11](https://linear.app/the-commit-company/issue/RAV-11/add-support-for-chat-in-desk-interface)</sub>

**Comment by nikkothari22** [comment]
@ajiragroup - yep this is exactly on our roadmap. We're currently cleaning up the codebase and improving performance/caching before we take this up. But this is definitely the end goal.

You can check out the video on #487 - we'll first start by adding sharing of documents to Raven via the timeline feature in Frappe (beside Send an Email, there would be Send a Raven).

Next, we'll work on making a small chat popup interface that lives on the Desk interface for quick chats like you mentioned. The Raven interface would be used for main chats or users who just want to use Raven. We're thinking about how we can build a reusable React/Vue/Vanilla JS library for this so that the Raven interface can be embedded into a Vue app (all new Frappe apps) and any custom app that users might have.

**Comment by ajiragroup** [comment]
I also suggest integrating Frappe Drive for file sharing and communication (optional/allow users to activate if drive is installed). If I have already stored a file in drive, there is no point in uploading 100's of MB of data again through Raven.

**Comment by nikkothari22** [comment]
> I also suggest integrating Frappe Drive for file sharing and communication (optional/allow users to activate if drive is installed). If I have already stored a file in drive, there is no point in uploading 100's of MB of data again through Raven.

Also in our pipeline - however we are waiting for Frappe Drive to be released and for us to try it out on v15. We have lots of plans, barely any bandwidth to work on them though 😅

**Comment by ajiragroup** [comment]
Yeah, same here. waiting for drive and many other apps release. No idea when that's going to happen !

**Comment by nikkothari22** [comment]
Adding list of tasks to do here:

- [x] Show a small panel on the side of the Desk interface to open Raven chat
- [x] On open, fetch all channels for the user (channels + DMs). Allow user to click on channel name to open the chat
- [x] Able to view text messages with formatting
- [x] Able to view images
- [x] Able to view videos
- [x] Able to view files 
- [ ] Able to send messages

Tasks that we will take up separately (will club some of them with the mobile app implementation)
- [ ] Able to react to messages
- [ ] Able to view reactions on a mesage
- [ ] Able to view a reply to a message
- [ ] Able to reply to a message
- [ ] Able to view linked document with message
- [ ] Able to send a message with the current document opened in the background
- [ ] Users can choose whether they want to enable chat inside Desk under Raven Users.

**Comment by nikkothari22** [comment]
Tracking remaining features on #647
